command/setup_consul.go

```diff
@@ -0,0 +1,485 @@
+
+  This command requires acl:write permissions for Consul and will respect
+  CONSUL_HTTP_TOKEN as well as CONSUL_HTTP_ADDR and other Consul-related
+  environment variables as documented in
+  https://developer.hashicorp.com/nomad/docs/runtime/environment#summary. 
+
+Setup Consul options:
+
+  -jwks-url
+    URL of the JWKS server Consul will contact in order to verify JWT
+    signatures. Defaults to http://localhost:4646/.well-known/jwks.json. 
+
+  -y
+    Automatically answers "yes" to all the questions, making the setup
+    non-interactive. Defaults to "false".
+
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (s *SetupConsulCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(s.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-jwks-url": complete.PredictAnything,
+			"-y":        complete.PredictSet("true", "false"),
+		})
+}
+
+func (s *SetupConsulCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (s *SetupConsulCommand) Synopsis() string { return "Setup a Consul cluster for Nomad integration" }
+
+// Name returns the name of this command.
+func (s *SetupConsulCommand) Name() string { return "setup consul" }
+
+// Run satisfies the cli.Command Run function.
+func (s *SetupConsulCommand) Run(args []string) int {
+
+	flags := s.Meta.FlagSet(s.Name(), FlagSetClient)
+	flags.Usage = func() { s.Ui.Output(s.Help()) }
+	flags.BoolVar(&s.autoYes, "y", false, "")
+	flags.StringVar(&s.jwksURL, "jwks-url", "http://localhost:4646/.well-known/jwks.json", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments.
+	if len(flags.Args()) != 0 {
+		s.Ui.Error("This command takes no arguments")
+		s.Ui.Error(commandErrorText(s))
+		return 1
+	}
+
+	var err error
+
+	s.Ui.Output(`
+This command will walk you through configuring all the components required for 
+Nomad workloads to authenticate themselves against Consul ACL using their 
+respective workload identities. 
+
+First we need to connect to Consul. 
+`)
+
+	cfg := api.DefaultConfig()
+	if !s.autoYes {
+		if !s.askQuestion(fmt.Sprintf("Is %s the correct address of your Consul cluster? [Y/n]", cfg.Address)) {
+			s.Ui.Warn(`
+Please set the CONSUL_HTTP_ADDR variable to your Consul cluster address and re-run the command.`)
+			return 0
+		}
+	}
+
+	// Get the Consul client.
+	s.client, err = api.NewClient(cfg)
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing Consul client: %s", err))
+		return 1
+	}
+
+	// check if we're connecting to Consul ent
+	if _, err := s.client.Operator().LicenseGet(nil); err == nil {
+		s.consulEnt = true
+	}
+
+	authMethodMsg := `
+We need to create two JWT auth methods: one for Nomad services, and one for
+Nomad tasks. The method for services will be called %v and the
+method for tasks %v, and they will both be of jwt type. They
+will share the following config:`
```

Comment:
This is a minor fit-and-finish issue, but these lines aren't evenly wrapped.

Reply:
That's for a reason. I actually calculated the characters of method names that get inserted in this message to get perfect alignment. (I am not joking)

Reply:
Ah, perfect! Ship it! :grinning: